### ISSUE TITLE
Include CUDA stack information in profile plots

### ISSUE THREAD

**Issue by pentschev** [dask/distributed#2578]
We're in the process of include low-level (e.g., C calls) profiling plots in https://github.com/dask/distributed/pull/2575. After that it may be useful to do the same with CUDA, we could add both host-side calls (e.g., memory allocation) and kernel calls.

This can be done with the help of [CUPTI - CUDA Profiling Tools Interface](https://docs.nvidia.com/cuda/cupti/index.html). Perhaps the C side could (and maybe, should) be part of a small library decoupled from this repository, just like https://github.com/numba/stacktrace.

**Comment by mrocklin** [comment]
cc @kkraus14 would this be helpful?  cc @sklam @seibert any thoughts?

**Comment by sklam** [comment]
On the GPU, I think you want to know:

* What the GPU is doing?  H<->D memcpy?  Kernel call?
* If kernel call? what's the symbol name of the kernel?

**Comment by pentschev** [comment]
> On the GPU, I think you want to know:
> 
> * What the GPU is doing?  H<->D memcpy?  Kernel call?
> * If kernel call? what's the symbol name of the kernel?

Exactly, those would be the most important information, there would be other calls that may be useful too, like `cudaStreamSynchronize()` and the likes.

**Comment by kkraus14** [comment]
I believe the libcudf developers would love this for understanding libcudf performance within a Dask-cudf workload.

**Comment by mrocklin** [comment]
Is there anyone that we know that would be willing to try it out?
